fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexQueryPlan.java

```diff
@@ -68,7 +68,7 @@ public int hashCode() {
     public int planHash(@Nonnull final PlanHashKind hashKind) {
         switch (hashKind) {
             case LEGACY:
-                return indexName.hashCode() + scanType.planHash(hashKind) + comparisons.planHash(hashKind) + sort.planHash() + (reverse ? 1 : 0);
+                return indexName.hashCode() + scanType.planHash(hashKind) + comparisons.planHash(hashKind) + (sort != null ? sort.planHash() : 0) + (reverse ? 1 : 0);
```

Comment:
There is no real LEGACY here as lucene is new. How about you just treat `LEGACY` and `STRUCTURAL_WITHOUT_LITERALS` the same,.

Reply:
In other words just more the `case LEGACY:` a few lines down.